File: src/main/java/io/github/daviddenton/metrique/metrics/MetricsClient.java
package io.github.daviddenton.metrique.metrics;

public interface MetricsClient {
    void decrement(MetricName name);

    void increment(MetricName name);

    void gauge(MetricName name, Long value);

    void time(MetricName name, Long value);

    void count(MetricName name, Long value);
}

File: src/main/java/io/github/daviddenton/metrique/metrics/MetricsClient.scala
package io.github.daviddenton.metrique.metrics

trait MetricsClient {
  def decrement(name: MetricName): Unit

  def increment(name: MetricName): Unit

  def gauge(name: MetricName, value: Long): Unit

  def time(name: MetricName, value: Long): Unit

  def count(name: MetricName, value: Long): Unit
}
